### DISCUSSION TITLE
VTX power increase after autolaunch is detected

### DISCUSSION THREAD

**Discussion by zoran-grujic** [iNavFlight/inav#6969] [Ideas]
## Current Behavior

If set, VTX will operate at low power before aircraft is armed. After arming VTX power level will be set according to settings.

## Desired Behavior

If autolaunch is permanently enabled (or enabled) switch to elevated VTX power AFTER the launch event has been detected.

## Suggested Solution

If aircraft is in AUTOLAUNCH mode maintain LOW VTX power until LAUNCH event is detected. Then switch to the set level.

## Who does this impact? Who is this for?

For large winged aircraft it is common that you arm (enable autolaunch) and then go to get bungee line, stretch it, and get in air. This procedure might take some time, and if you have some extra problems even minutes. So the VTX might get hot due no proper airflow with all the consequences. Before the airplane is actually flying low VTX power is sufficient

**Comment by DzikuVx** [comment]
As an alternative I suggest using Programming Framework and setting higher output power when distance to home is > 50m for example. There are several tutorials how to implement a function like that

**Comment by P-I-Engineer** [comment]
that programming framework really is awesome, it's like having opentx freedom inside your FC.

**Comment by zoran-grujic** [comment]
Yes, the programming interface is useful and should be used for all nonstandard operations. But I think changing VTX power should be a standard part of INAV. 

I managed to use the programming interface for VTX control as function of home distance. But for majority of the people out there this is ADVANCED MAGIC. I did test flight and my logic was running fine except in the case when I am very high above arm position but very close in terms of GPS distance. Now I realized that there is also 3D home distance flight value I could use... 
Would be nice trough to have NOT operation on logic conditions when used as operands.

I wish to drop the proposal on the top. While useful, and probably simple to implement, it is not THE solution.
INAV should have a table of VTX power levels as function of the 3D distance. Something like:
<50 m -> Level 1
50 - 200 m -> Level 2
200 - 500 m -> Level 3
etc.

where user can enter distances and power levels with ZERO programming and logic knowledge.
This is my solution:

# logic
logic 0 1 -1 3 2 31 0 50 0
logic 1 1 -1 3 2 31 0 200 0
logic 2 1 -1 3 2 31 0 2000 0
logic 3 1 0 25 0 1 0 0 0
logic 4 1 -1 12 4 2 0 0 0
logic 5 1 4 25 0 4 0 0 0
logic 6 1 -1 12 4 0 0 0 0
logic 7 1 -1 7 4 6 4 1 0
logic 8 1 7 25 0 2 0 0 0
logic 9 1 -1 12 4 1 0 0 0
logic 10 1 -1 7 4 9 4 2 0
logic 11 1 10 25 0 3 0 0 0

**Comment by wx4cb** [comment]
this can already be done as per logic conditions as you already know, but however i don't think that such an advanced function should be "standard" due to the fact that it can EASILY be misconfigured. there is already a "low power on disarm" (or words to that effect) function in the configuration tab that is pretty much the best way to do it imho.
